From f22458a7b9b48da64977b6c8c83ae2657b3579eb Mon Sep 17 00:00:00 2001
Message-Id: <f22458a7b9b48da64977b6c8c83ae2657b3579eb.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Mon, 7 Jul 2014 15:27:15 +0100
Subject: [PATCH 141/312] FOR_UPSTREAM [VPG]: drm/i915: Add
 WaDisableInstructionShootdown:bdw

Instruction Shootdown in ROW_CHICKEN must be disabled.

Issue: VIZ-4086
Change-Id: Id25fa494d9588bf7e24dd80b52f3f68ce453f82d
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    1 +
 drivers/gpu/drm/i915/intel_pm.c |    4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index d0abceb..7b72b08 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -5685,6 +5685,7 @@ enum punit_power_well {
 #define   GEN7_PSD_SINGLE_PORT_DISPATCH_ENABLE	(1<<3)
 
 #define GEN8_ROW_CHICKEN		0xe4f0
+#define   INSTRUCTION_SHOOTDOWN_DISABLE	(1<<9)
 #define   PARTIAL_INSTRUCTION_SHOOTDOWN_DISABLE	(1<<8)
 #define   STALL_DOP_GATING_DISABLE		(1<<5)
 
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index f2ce500..912b9ed 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5527,8 +5527,10 @@ static void gen8_init_clock_gating(struct drm_device *dev)
 		   _MASKED_BIT_ENABLE(PARTIAL_INSTRUCTION_SHOOTDOWN_DISABLE));
 
 	/* WaDisableThreadStallDopClockGating:bdw */
-	/* FIXME: Unclear whether we really need this on production bdw. */
+	/* WaDisableInstructionShootdown:bdw */
+	/* FIXME: Unclear whether we really need these on production bdw. */
 	I915_WRITE(GEN8_ROW_CHICKEN,
+		   _MASKED_BIT_ENABLE(INSTRUCTION_SHOOTDOWN_DISABLE) |
 		   _MASKED_BIT_ENABLE(STALL_DOP_GATING_DISABLE));
 
 	/*
-- 
1.7.9.5

